lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb

```diff
@@ -497,8 +499,11 @@ def add_resources_bundle_targets(file_accessors)
                 target.user_build_configurations.each do |bc_name, type|
                   resource_bundle_target.add_build_configuration(bc_name, type)
                 end
-                resource_bundle_target.deployment_target = deployment_target
-
+                resource_bundle_target.deployment_target = if file_accessor.spec.non_library_specification?
```

Comment:
is this covered by tests?

Reply:
eeeeeh kind of tbh but I will add one.

Reply:
@amorde _really_ thank you for pushing here as I discovered this is not working properly because of my test.

Reply:
yay for tests :)